Fix an issue where empty keys were added in app bulk translations

AmitPhulera opened the pull request:
## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
<!-- Where applicable, describe user-facing effects and include screenshots. -->
We had an issue with our bulk translations logic where we were allowing empty keys if they were present in uploaded Excel file. This empty key was causing issues with ES. (See https://github.com/dimagi/commcare-hq/pull/35889).

The issue is demonstrated by the test failure [here](https://github.com/dimagi/commcare-hq/actions/runs/13840998556/job/38728172429#step:5:673) 

## Safety Assurance
Does not change any functionality, fixes an issue, also have tests for the fix.
### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
Tests are added
### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

n/a

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

AmitPhulera committed:
add test to demonstrate empty key in translations

AmitPhulera committed:
add fix for the empty key appearing in app translations

AmitPhulera committed:
lint

millerdev commented on corehq/apps/app_manager/ui_translations/ui_translations.py:
```diff
@@ -68,6 +68,10 @@ def _get_params_text(params):
 
     for row in translations:
+        if row["property"] == "":
+            warnings.append("Property '" + row["property"] + "' is empty, "
+                            "We did not add it to the translations")
```
Should this warning message be translated? A few lines below (line 87 and below) error and warning messages are translated.

AmitPhulera replied:
https://github.com/dimagi/commcare-hq/pull/35969/commits/35d0bfe32e885f4a397d9afaebb74c4d1a21a0ea

AmitPhulera committed:
translate error messages

AmitPhulera committed:
fix failing test

when there is an empty key format returns empty key which does not appear as '', so directly using '' to display empty property is good idea

millerdev commented on corehq/apps/app_manager/ui_translations/ui_translations.py:
```diff
@@ -68,10 +69,14 @@ def _get_params_text(params):
 
     for row in translations:
+        if row["property"] == "":
+            warnings.append(_("Property '' is empty, "
+                            "We did not add it to the translations"))
```
nit
```suggestion
            warnings.append(_("Property '' is empty. "
                            "We did not add it to the translations"))
```

millerdev commented on corehq/apps/app_manager/ui_translations/ui_translations.py:
```diff
@@ -68,10 +69,14 @@ def _get_params_text(params):
             # Add a warning for  unknown properties, but still add them to the translation dict
-            warnings.append("Property '" + row["property"] + "' is not a known CommCare UI string, "
-                            "but we added it anyway.")
+            warnings.append(_("Property {prop} is not a known CommCare UI string, "
+                            "but we added it anyway.").format(prop=row["property"]))
```
nit: Would it make sense to put the property name in quotes? Imagine a property named 'name', which would result in an ambiguous and possibly confusing message like "Property name is not a known CommCare UI string..."
```suggestion
            warnings.append(_("Property '{prop}' is not a known CommCare UI string, "
                            "but we added it anyway.").format(prop=row["property"]))
```

AmitPhulera committed:
Apply suggestions from code review

Co-authored-by: Daniel Miller <dmiller@dimagi.com>

AmitPhulera committed:
fix tests
